src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetDepartmentHandlerBase.cs

```diff
@@ -1,66 +1,67 @@
 ﻿using Fusion.Integration;
 using Fusion.Integration.LineOrg;
 using Fusion.Integration.Profile;
+using Fusion.Integration.Roles;
 using Fusion.Resources.Database;
 using Fusion.Resources.Database.Entities;
 using Microsoft.EntityFrameworkCore;
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Fusion.Resources.Domain
 {
     public class DepartmentHandlerBase
     {
-        protected readonly ResourcesDbContext db;
+        private readonly IFusionRolesClient rolesClient;
         protected readonly ILineOrgResolver lineOrgResolver;
         protected readonly IFusionProfileResolver profileResolver;
-        public DepartmentHandlerBase(ResourcesDbContext db, ILineOrgResolver lineOrgResolver, IFusionProfileResolver profileResolver)
+
+        public DepartmentHandlerBase(IFusionRolesClient rolesClient, ILineOrgResolver lineOrgResolver, IFusionProfileResolver profileResolver)
         {
-            this.db = db;
+            this.rolesClient = rolesClient;
             this.lineOrgResolver = lineOrgResolver;
             this.profileResolver = profileResolver;
         }
 
         internal async Task ExpandDelegatedResourceOwner(QueryDepartment department, CancellationToken cancellationToken)
         {
-            var delegatedResourceOwners = await db.DepartmentResponsibles
-                            .Where(r => r.DepartmentId == department.DepartmentId)
-                            .ToListAsync(cancellationToken);
-
+            var delegatedResourceOwners = await rolesClient.GetRolesAsync(q => q
+                   .WhereScopeValue(department.DepartmentId))
```

Comment:
You can add a couple of more restrictions just in case. The scope value is not unique (can be used elsewhere)
.WhereScopeType("OrgUnit") 
.WhereRoleName(AccessRoles.ResourceOwner)